### PR TITLE
Malus: Implement storing invalid chunks

### PR DESCRIPTION
**Pull request by Lldenaurois**
This PR implements a first attempt at intercepting AvailabilityStoreMessages and writing invalid chunks into the availability store for every `StoreAvailableData` message intercepted by the MessageInterceptor.

### CODE REVIEW

File: node/malus/src/variants/store_malicious_available_data.rs [hunk 1]
```diff
@@ -0,0 +1,353 @@
...
+		prepared_overseer_builder(args)?
+			.replace_availability_store(|av| InterceptedSubsystem::new(av, filter))
+			.build_with_connector(connector)
+			.map_err(|e| e.into())
+	}
+}
+
+/// Unix time wrapper with big-endian encoding.
+#[derive(Debug, Clone, Copy, PartialEq, PartialOrd, Eq, Ord)]
+struct BETimestamp(u64);
```
Comment (sandreim) [inline node/malus/src/variants/store_malicious_available_data.rs:217]: This duplicates https://github.com/paritytech/polkadot/blob/master/node/core/av-store/src/lib.rs#L78. We can reuse that to reduce the amount of code written in malus.

File: node/malus/src/variants/store_malicious_available_data.rs [hunk 2]
```diff
@@ -0,0 +1,353 @@
+// Copyright 2021 Parity Technologies (UK) Ltd.
```
Comment (sandreim) [inline node/malus/src/variants/store_malicious_available_data.rs:1]: Can we avoid duplicating this file: https://github.com/paritytech/polkadot/blob/master/node/core/av-store/src/lib.rs ?

File: node/malus/src/malus.rs [hunk 3]
```diff
@@ -63,10 +63,10 @@ impl MalusCli {
 	/// Launch a malus node.
 	fn launch(self) -> eyre::Result<()> {
 		match self.variant {
-			NemesisVariant::BackGarbageCandidate(cmd) =>
-				polkadot_cli::run_node(run_cmd(cmd), BackGarbageCandidate)?,
+			NemesisVariant::StoreMaliciousAvailableData(cmd) =>
+				polkadot_cli::run_node(run_cmd(cmd), StoreMaliciousAvailableDataWrapper)?,
 			NemesisVariant::SuggestGarbageCandidate(cmd) =>
-				polkadot_cli::run_node(run_cmd(cmd), SuggestGarbageCandidate)?,
+				polkadot_cli::run_node(run_cmd(cmd), BackGarbageCandidateWrapper)?,
```
Comment (sandreim) [inline node/malus/src/malus.rs:69]: Do we want to maybe combine these eventually? So we can run tests with malicious stored available data and dispute valid candidates for example.

**Review by sandreim — COMMENTED** [review]
Took a first pass, will take a second one and pay more attention to details vs reference impl.

File: node/malus/src/variants/dispute_valid_candidates.rs [hunk 4]
```diff
@@ -104,17 +102,10 @@ impl OverseerGen for DisputeValidCandidates {
...
 		let filter = ReplaceApprovalsWithDisputes;
 
 		prepared_overseer_builder(args)?
-			.replace_candidate_backing(move |cb| {
-				InterceptedSubsystem::new(
-					CandidateBackingSubsystem::new(spawner, crypto_store_ptr, cb.params.metrics),
-					filter,
-				)
-			})
+			.replace_candidate_backing(move |cb| InterceptedSubsystem::new(cb, filter))
```
Comment (drahnr) [inline node/malus/src/variants/dispute_valid_candidates.rs:108]: :+1:

**Comment by drahnr** [comment]
@sandreim does this bring any additional coverage to what we have? If not let's close this

**Pull request closed without merging**